crates/bestool/src/actions/tamanu/backup.rs

```diff
@@ -0,0 +1,138 @@
+use std::{fs, path::Path};
+
+use chrono::Local;
+use clap::Parser;
+use miette::{Context as _, IntoDiagnostic as _, Result};
+use tracing::{debug, info};
+
+use crate::actions::{
+	tamanu::{
+		config::{merge_json, package_config},
+		find_package, find_postgres_bin, find_tamanu, TamanuArgs,
+	},
+	Context,
+};
+
+/// Dump a local Tamanu DB using "pg_dump".
+///
+/// This finds the database from the Tamanu's configuration. The output will be written to a file
+/// "{current_datetime}-{host_name}-{database_name}.dump".
+///
+/// By default, this excludes tables "sync_snapshots.*" and "fhir.jobs".
+#[derive(Debug, Clone, Parser)]
+pub struct BackupArgs {
+	/// The compression level to use.
+	///
+	/// This is simply passed to the "--compress" option of "pg_dump".
+	#[arg(long, default_value_t = 3)]
+	compression_level: u32,
+
+	/// The destination directory the output will be written to.
+	#[cfg_attr(windows, arg(long, default_value = r"C:\Backup"))]
+	#[cfg_attr(not(windows), arg(long, default_value = "/backup"))]
+	write_to: String,
+
+	/// TODO:
+	#[arg(long)]
+	then_copy_to: Option<String>,
+
+	/// Enable the lean backup
+	///
+	/// The lean backup excludes more tables: "logs.*", "reporting.*" and "public.attachments".
+	#[arg(long, default_value_t = false)]
+	lean: bool,
+}
+
+#[derive(serde::Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct TamanuConfig {
+	canonical_host_name: String,
+	db: TamanuDb,
+}
+
+#[derive(serde::Deserialize, Debug)]
+struct TamanuDb {
+	name: String,
+	username: String,
+	password: String,
+}
+
+pub async fn run(ctx: Context<TamanuArgs, BackupArgs>) -> Result<()> {
+	// TODO: # Use two processor cores at most
```

Comment:
that's actually not worth porting as it's ineffectual: it only affects this process, but the bulk of the work is done by postgres daemon